lib/AST/Decl.cpp

```diff
@@ -4288,7 +4288,11 @@ static Type computeNominalType(NominalTypeDecl *decl, DeclTypeKind kind) {
   // If `decl` is a nested type, find the parent type.
   Type ParentTy;
   DeclContext *dc = decl->getDeclContext();
-  if (!isa<ProtocolDecl>(decl) && dc->isTypeContext()) {
+  bool isObjCProtocol = isa<ProtocolDecl>(decl) && decl->hasClangNode();
```

Comment:
Is there any harm in doing this for all protocols?

Reply:
Logically it shouldn't be but who knows what side-effects calling stuff on `dc` is going to have, I'm just not brave enough :)

Reply:
Alright then, let's remove that check completely!

Reply:
Done!